.github/workflows/call-deploy-book.yml

```diff
@@ -0,0 +1,23 @@
+name: call-deploy-book
+
+on:
+  push:
+    branches:
+    - '**'
+    # If your git repository has the Jupyter Book within some-subfolder next to
+    # unrelated files, you can make this run only if a file within that specific
+    # folder has been modified.
+    #
+    # paths:
+    # - some-subfolder/**
+  workflow_dispatch:
+
+jobs: 
+  call-workflow:
+    uses: TeachBooks/template/.github/workflows/deploy-book-ghpages.yml@main
+    secrets: inherit
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+
```

Comment:
I think the config vars are passed either by default, or with secrets. I will try removing `secrets: inherit` to see if we can get away without it

Reply:
But those configuration variables are not defined in the `call` workflow now right? So how does it know what to pass through?

Reply:
Yes, the config variables are not defined in the call workflow file, but they need not be -- they are defined on the repo-level like this: https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repositor then *all of them* are passed along to the called workflow, where they are set to the *env* variables.

Reply:
Sure, but if the `deploy-book-ghpages.yml` workflow is not in the repo, those variables should be defined in the `call-deploy-book.yml` workflow right?

Reply:
Ah I see, however, those variables shouldn't be the same for all repos: ideally every book could customize those in `call-workflow.yml` while keeping the actual called workflow file the same. Would that be an option?

Reply:
Yes, the config vars are repo-specific.

It is an option, however there is a better one. If users were to specify the variable values in the `.yml` file, they would have to do so in all branches at once (because a workflow is triggered on push *per branch*, so the set value would be *per branch*). The better option is, users (with appropriate access) set the var values for their whole repo at once in the GitHub web GUI with this https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository

Reply:
I tested it in the sandbox https://github.com/TeachBooks/Sandbox/settings/variables/actions works brilliantly! Thanks Severin!